### PR TITLE
build: update to TypeScript 5.5 stable

### PR DESCRIPTION
**Pull request by crisbeto**
Updates the repo to the stable version of TypeScript 5.5. I'm holding off on updating the new projects to it due to https://github.com/angular/angular/pull/56358.

### CODE REVIEW

**Review by alan-agius4 — APPROVED** [review]

**Comment by crisbeto** [comment]
I mentioned it in the commit message: I'll hold off on switching new projects to it for now until we can land https://github.com/angular/angular/pull/56358. Without it there's an edge case where users can see a very cryptic TS error.

**Pull request merged**
